tests/test_oc4ids_tranforms.py

```diff
@@ -0,0 +1,1332 @@
+    assert "funder" in output["parties"][0]["roles"]
+
+
+def test_cost_estimate():
+
+    releases = [
+        {
+            "ocid": "ocds-213czf-1",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "tender": {"status": "planning", "value": {"amount": 1}},
+        },
+        {
+            "ocid": "ocds-213czf-1",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "tender": {"status": "planning", "value": {"amount": 10}},
+        },
+    ]
+
+    output = transforms._run_transforms(
+        releases, "1", transforms=[transforms.contracting_process_setup, transforms.cost_estimate],
+    )
+
+    assert output["contractingProcesses"][0]["summary"]["tender"]["costEstimate"] == {"amount": 10}
+
+    # reverse releases
+    releases = releases[::-1]
+
+    output = transforms._run_transforms(
+        releases, "1", transforms=[transforms.contracting_process_setup, transforms.cost_estimate],
+    )
+
+    assert output["contractingProcesses"][0]["summary"]["tender"]["costEstimate"] == {"amount": 1}
+
+    releases.append(
+        {
+            "ocid": "ocds-213czf-1",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "tender": {"value": {"amount": 100}},
+        }
+    )
+
+    # last releases is not planning
+    output = transforms._run_transforms(
+        releases, "1", transforms=[transforms.contracting_process_setup, transforms.cost_estimate],
+    )
+    assert output["contractingProcesses"][0]["summary"]["tender"]["costEstimate"] == {"amount": 1}
+
+
+def test_contract_title():
+
+    releases = [
+        {
+            "ocid": "ocds-213czf-1",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "contracts": [{"title": "a"}],
+            "awards": [{"title": "b"}],
+            "tender": {"title": "c"},
+        }
+    ]
+
+    output = transforms._run_transforms(
+        releases, "1", transforms=[transforms.contracting_process_setup, transforms.contract_title],
+    )
+
+    assert output["contractingProcesses"][0]["summary"]["title"] == "a"
+
+    # with second contract we do not use contract title
+    releases[0]["contracts"].append({"title": "a"})
+
+    output = transforms._run_transforms(
+        releases, "1", transforms=[transforms.contracting_process_setup, transforms.contract_title],
+    )
+
+    assert output["contractingProcesses"][0]["summary"]["title"] == "b"
+
+    # with second awards we also do not use award title
+    releases[0]["awards"].append({"title": "b"})
+
+    output = transforms._run_transforms(
+        releases, "1", transforms=[transforms.contracting_process_setup, transforms.contract_title],
+    )
+
+    assert output["contractingProcesses"][0]["summary"]["title"] == "c"
+
+
+def test_supplier():
```

Comment:
Should the test cover an input with multiple contracting processes, with suppliers declared in each? Also need to handle clashing `parties.id`s